langserver/config.go

```diff
@@ -77,6 +68,11 @@ func (c *Config) Validate() error {
 			return err
 		}
 	}
+
+	if !regexp.MustCompile("(text|json)?").MatchString(c.LogFormat) {
```

Comment:
Looks like a sort of enum for me :)
```go
type LogFormat string

const(
   JSONFormat LogFormat = "json"
   TextFormat LogFormat = "text"
)

var mapLogFormat = map[LogFormat]bool{
   JSONFormat : true,
   TextFormat : true,
}

[....]
if len(c.LogFormat) > 0 {
  if !mapLogFormat[c.LogFormat] {
      return fmt.Errorf(...)
  }
} else {
   // default value
    c.LogFormat = TextFormat
}
```
Doing it through a regexp is not really the elegant way to describe an enum.

And if you are using the method `MustCompile` then you should declare a global variable. Like that the binary will crash immediately when it is started.

Like that it will crash only if you are parsing the config. And for some cases (not here I know) the code won't be executed

Reply:
Right, an enum approach is better here.